python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -265,9 +266,7 @@ def sort_values(
     df3.divisions = (None,) * (df3.npartitions + 1)
 
     # Step 3 - Return final sorted df
-    df4 = df3.map_partitions(
-        M.sort_values, by, ascending=ascending, na_position=na_position
-    )
+    df4 = df3.map_partitions(sort_function, **sort_function_kwargs)
```

Comment:
Something feels off here. We are requiring that the user specify `sort_function`, but the API makes it seem optional.  I worry that we are now silently ignoring `acsending` and `na_position` (and maybe even `by`?).

What if down-stream users are implementing code with `sorting.sort_values` directly?  I don't think that is good/recommended practice, but the API we are changing seems "public" to me (making this a breaking change).

Perhaps a simpler (non-breaking) solution would be to remove most of the changes from `DataFrame.sort_values`, pass through `sort_function` and `sort_function_kwargs` into here, and implement the `sort_function`/`sort_function_kwargs` default logic here (in `sorting.sort_values`).  Does this seem reasonable?

Reply:
That makes sense and is a valid concern - my only comment is that we ideally still want to allow for custom sorting functions in the `npartitions == 1` case that is handled directly in `DataFrame.sort_values`, so I think it might also make sense to move the following logic:

```python
        if self.npartitions == 1:
            df = self.map_partitions(sort_function, **sort_kwargs)
```

into `sorting.sort_values` as well, unless there's a reason that's not immediately obvious to me why we would want to keep the single partition case separate?

Also noting that this is also a concern for the upstream implementation of this, so depending on what we decide on here I will open up a follow up PR to address this in Dask.

Reply:
>Also noting that this is also a concern for the upstream implementation of this, so depending on what we decide on here I will open up a follow up PR to address this in Dask.

Good point!  I definitely like the simplification you made here.  So it probably makes sense to do something similar upstream.